exporter/chronicleexporter/marshal.go

```diff
@@ -73,64 +72,89 @@ func newProtoMarshaler(cfg Config, teleSettings component.TelemetrySettings, lab
 }
 
 func (m *protoMarshaler) MarshalRawLogs(ctx context.Context, ld plog.Logs) ([]*api.BatchCreateLogsRequest, error) {
-	rawLogs, err := m.extractRawLogs(ctx, ld)
+	rawLogs, namespace, ingestionLabels, err := m.extractRawLogs(ctx, ld)
 	if err != nil {
 		return nil, fmt.Errorf("extract raw logs: %w", err)
 	}
-
-	return m.constructPayloads(rawLogs), nil
+	return m.constructPayloads(rawLogs, namespace, ingestionLabels), nil
 }
 
-func (m *protoMarshaler) extractRawLogs(ctx context.Context, ld plog.Logs) (map[string][]*api.LogEntry, error) {
+func (m *protoMarshaler) extractRawLogs(ctx context.Context, ld plog.Logs) (map[string][]*api.LogEntry, map[string]string, map[string][]*api.Label, error) {
 	entries := make(map[string][]*api.LogEntry)
-
+	namespaceMap := make(map[string]string)
+	ingestionLabelsMap := make(map[string][]*api.Label)
 	for i := 0; i < ld.ResourceLogs().Len(); i++ {
 		resourceLog := ld.ResourceLogs().At(i)
 		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
 			scopeLog := resourceLog.ScopeLogs().At(j)
 			for k := 0; k < scopeLog.LogRecords().Len(); k++ {
 				logRecord := scopeLog.LogRecords().At(k)
-				rawLog, logType, err := m.processLogRecord(ctx, logRecord, scopeLog, resourceLog)
+				rawLog, logType, namespace, ingestionLabels, err := m.processLogRecord(ctx, logRecord, scopeLog, resourceLog)
 				if err != nil {
 					m.teleSettings.Logger.Error("Error processing log record", zap.Error(err))
 					continue
 				}
-
 				if rawLog == "" {
 					continue
 				}
-
 				var timestamp time.Time
 				if logRecord.Timestamp() != 0 {
 					timestamp = logRecord.Timestamp().AsTime()
 				} else {
 					timestamp = logRecord.ObservedTimestamp().AsTime()
 				}
-
 				entry := &api.LogEntry{
 					Timestamp:      timestamppb.New(timestamp),
 					CollectionTime: timestamppb.New(logRecord.ObservedTimestamp().AsTime()),
 					Data:           []byte(rawLog),
 				}
 				entries[logType] = append(entries[logType], entry)
+				if namespace != "" {
+					namespaceMap[logType] = namespace
+				}
+				if len(ingestionLabels) != 0 {
+					if ingestionLabelsMap[logType] == nil {
+						ingestionLabelsMap[logType] = make([]*api.Label, 0)
+					}
+					for _, v := range ingestionLabels {
+						ingestionLabelsMap[logType] = append(ingestionLabelsMap[logType], v)
+					}
+				}
```

Comment:
I think some more thought needs to go into how these labels get added. Currently we append all labels present on a log record to the overarching map, regardless of if they're already present. This causes repeat entries if there are multiple log records with the same labels. You can add this test case to the test `TestProtoMarshaler_MarshalRawLogs` in `marshal_test.go` test to see this behavior:
```go
{
			name: "Multiple log records with attributes",
			cfg: Config{
				CustomerID:      uuid.New().String(),
				LogType:         "WINEVTLOG",
				Namespace:       "test",
				RawLogField:     "body",
				OverrideLogType: false,
			},
			labels: []*api.Label{},
			logRecords: func() plog.Logs {
				logs := plog.NewLogs()
				record1 := logs.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
				record1.Body().SetStr("First log message")
				record1.Attributes().FromRaw(map[string]any{`chronicle_ingestion_label["key1"]`: "value1", `chronicle_ingestion_label["key2"]`: "value2"})
				record2 := logs.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
				record2.Body().SetStr("Second log message")
				record2.Attributes().FromRaw(map[string]any{`chronicle_ingestion_label["key1"]`: "value1", `chronicle_ingestion_label["key2"]`: "value2"})
				return logs
			},
			expectations: func(t *testing.T, requests []*api.BatchCreateLogsRequest) {
				require.Len(t, requests, 1, "Expected a single batch request")
				batch := requests[0].Batch
				// verify batch source labels
				require.Len(t, batch.Source.Labels, 2)
				require.Len(t, batch.Entries, 2, "Expected two log entries in the batch")
				// Verifying the first log entry data
				require.Equal(t, "First log message", string(batch.Entries[0].Data))
				// Verifying the second log entry data
				require.Equal(t, "Second log message", string(batch.Entries[1].Data))
			},
		},
```